lib/set-http-code.js

```diff
@@ -0,0 +1,30 @@
+// Copyright IBM Corp. 2012,2017. All Rights Reserved.
+// Node module: loopback-connector-mysql
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+'use strict';
+
+var _ = require('lodash');
+
+/*!
+ * Translate MySQL error codes into HTTP Errors.
+ * If an error would be better represented by a code that isn't 500,
+ * add your logic to this function!
+ */
+module.exports = function(err) {
+  if (!err) {
+    return;
+  } else if (!(err instanceof Error)) {
+    err = new Error(err); // Sucks that we weren't given an error object...
+  }
+
+  // Convert errors into status codes here.
+  if (_.includes(err.message, 'ER_DUP_ENTRY')) {
+    err.statusCode = 422;
+  } else {
+    // If all else fails, call it a 500.
+    err.statusCode = 500;
```

Comment:
If I understand correctly, the idea to is to add more error types as applicable, but for now all else will be treated as 500.